models/comments.models.js

```diff
@@ -31,4 +31,23 @@ function insertCommentByArticleId(article_id,username,body){
         });
 
 }
-module.exports = { selectCommentsByArticleId,insertCommentByArticleId };
+
+
+//task 9
+function selectCommentsByCommentsId(comment_id){
+
+       if (isNaN(comment_id)) {
+          return Promise.reject({ status: 400, msg: 'Bad request: invalid comment_id' });
+        }
+
+return db.query (`DELETE FROM comments WHERE comment_id =$1 RETURNING *;`,[comment_id])
+.then((result)=>{
+    if (result.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: 'Comment not found' });
+    }
+    return result.rows[0];
+
+})
```

Comment:
if comment_id is not a number PSQL would throw an error so your if block isn't necessary.
Instead investigate the PSQL error by console.logging the err in the controller catch block and create a middleware errorhandler function in the app that would deal with this PSQL error, refer to the NC notes